text/3771-demote-i686-pc-windows-gnu.md

```diff
@@ -0,0 +1,172 @@
+
+## Target Tier Policy Requirements
+
+With this knowledge, we can look at the [Tier 1 requirements](https://doc.rust-lang.org/1.84.1/rustc/target-tier-policy.html#tier-1-target-policy) of the target tier policy to check whether they are fulfilled.
+
+> *Tier 1 targets must have substantial, widespread interest within the developer community, and must serve the ongoing needs of multiple production users of Rust across multiple organizations or projects.*
+
+While this cannot be quantified precisely, the download counts suggest that this target is less popular than some other Tier 2 targets like FreeBSD.
+Therefore, we are going to treat this as false.
+
+> The target maintainer team must include at least 3 developers.
+
+This is not the case at all. There is currently no maintainer team.
+Though we should note that this is currently also true for many other Tier 1 targets, as this is a new rule not upheld everywhere yet.
+But experience tells that it is highly unlikely that 3 maintainers for 32 bit Windows GNU will be found.
+
+> The target must build and pass tests reliably in CI, for all components that Rust's CI considers mandatory.
+
+As mentioned above, there are issues and it does cause a fair share of problems.
+
+> The target must not disable an excessive number of tests or pieces of tests in the testsuite in order to do so. This is a subjective requirement.
+
+A fair amount of tests are disabled with an open issue with no comments.
+I would say that it is on the edge of being excessive, not quite having reached that amount (but it is likely that will be reached eventually).
+
+> The target must provide as much of the Rust standard library as is feasible and appropriate to provide [...].
+
+Windows is well-supported in the standard library.
+
+> Building the target and running the testsuite for the target must not take substantially longer than other targets
+
+Building `i686-pc-windows-gnu` is reasonably fast.
+
+> If running the testsuite requires additional infrastructure
+
+GitHub Actions has Windows support, which is used for `i686-pc-windows-gnu` (on a 64-bit host), no external infrastructure is required.
+
+> Tier 1 targets must not have a hard requirement for signed, verified, or otherwise "approved" binaries.
+
+There are no such requirements.
+
+> All requirements for tier 2 apply.
+
+We will not go through Tier 2 requirements here, but they are, apart from the (less strict than Tier 1) maintainer requirements, fulfilled.
+When the maintainer requirements are enforced more strictly in the future, `i686-pc-windows-gnu` (and `x86_64-pc-windows-gnu` as well) may be demoted further if no maintainers are found.
+
+## Conclusion
+
+Given the usage count and lack of maintenance leading to more than one requirement not being fulfilled, it becomes clear that `i686-pc-windows-gnu` is not worthy of being a Tier 1 target and is already getting much worse support than expected from a Tier 1 target.
```

Comment:
Remark: arguably, it also imposes a maintenance burden on compiler contributors because if a test fails on just `i686-mingw` (32-bit windows-gnu) but not 64-bit windows-{msvc,gnu} it's very annoying to bless that.